sound/soc/sof/intel/hda-dai.c

```diff
@@ -136,7 +77,12 @@ static int hda_link_hw_params(struct snd_pcm_substream *substream,
 	struct hdac_ext_link *link;
 	int stream_tag;
 
-	link_dev = snd_soc_dai_get_dma_data(dai, substream);
+	link_dev = snd_hdac_ext_stream_assign(bus, substream,
+					      HDAC_EXT_STREAM_TYPE_LINK);
+	if (!link_dev)
+		return -EBUSY;
```

Comment:
doesn't seem quite right:
````c
hdac_ext_link_stream_assign(struct hdac_bus *bus,
				struct snd_pcm_substream *substream)
{
	struct hdac_ext_stream *res = NULL;
	struct hdac_stream *stream = NULL;

	if (!bus->ppcap) {
		dev_err(bus->dev, "stream type not supported\n");
		return NULL;
	}
````
-EINVAL?

Reply:
Oh yes, should be -EINVAL. will change it .